handsontable/src/plugins/base/__tests__/base.unit.js

```diff
@@ -140,6 +140,52 @@ describe('BasePlugin', () => {
 
       expect(plugin.getSetting('cells')).toEqual([3, 4, 5]);
     });
+
+    it('should return object with defaults and plugin settings', () => {
```

Comment:
Can I ask you to add a test that would cover the case where you are able to get the nested plugin setting using the dot notation (`hot.getPlugin('dialog').getSetting('a11y.role')`)?